lib/internal/test_runner/tests_stream.js

```diff
@@ -27,6 +27,10 @@ class TestsStream extends Readable {
     }
   }
 
+  begin(nesting, file, testNumber, name, details, directive) {
+    this.#emit('test:begin', { __proto__: null, name, nesting, file, testNumber, details, ...directive });
```

Comment:
Just throwing an idea out there: having both `begin` and `start` is confusing. I could also picture people asking for more of these lifecycle events. I would namespace this event further and name it `test:lifecycle:run`, where `lifecycle` can be used to indicate what is happening in the test runner and is less about reporting results.